packages/dd-trace/src/plugins/util/ci.js

```diff
@@ -29,45 +68,82 @@ module.exports = {
+
+      const tags = {
         [CI_PIPELINE_ID]: BUILD_TAG,
-        [CI_PIPELINE_NAME]: JOB_NAME,
         [CI_PIPELINE_NUMBER]: BUILD_NUMBER,
         [CI_PIPELINE_URL]: BUILD_URL,
         [CI_PROVIDER_NAME]: 'jenkins',
-        [CI_WORKSPACE_PATH]: WORKSPACE,
-        [GIT_BRANCH]: JENKINS_GIT_BRANCH,
         [GIT_COMMIT_SHA]: JENKINS_GIT_COMMIT,
-        [GIT_REPOSITORY_URL]: JENKINS_GIT_REPOSITORY_URL
+        [DEPRECATED_GIT_COMMIT_SHA]: JENKINS_GIT_COMMIT,
+        [GIT_REPOSITORY_URL]: filterSensitiveInfoFromRepository(JENKINS_GIT_REPOSITORY_URL)
+      }
+
+      const isTag = JENKINS_GIT_BRANCH && JENKINS_GIT_BRANCH.includes('tags')
+      const finalRefKey = isTag ? GIT_TAG : GIT_BRANCH
+      const ref = normalizeRef(JENKINS_GIT_BRANCH)
+
+      tags[finalRefKey] = ref
+
+      if (WORKSPACE) {
+        tags[CI_WORKSPACE_PATH] = resolveTilde(WORKSPACE)
+      }
+      let finalPipelineName = ''
+      if (JOB_NAME) {
+        // Job names can contain parameters, e.g. jobName/KEY1=VALUE1,KEY2=VALUE2/branchName
+        const jobNameAndParams = JOB_NAME.split('/')
+        if (jobNameAndParams.length > 1 && jobNameAndParams[1].includes('=')) {
+          finalPipelineName = jobNameAndParams[0]
+        } else {
+          finalPipelineName = JOB_NAME.replace(`/${ref}`, '')
+        }
+        tags[CI_PIPELINE_NAME] = finalPipelineName
       }
+
+      return tags
     }
 
     if (env.GITLAB_CI) {
       const {
         CI_PIPELINE_ID: GITLAB_PIPELINE_ID,
         CI_PROJECT_PATH,
         CI_PIPELINE_IID,
-        CI_PIPELINE_URL,
+        CI_PIPELINE_URL: GITLAB_PIPELINE_URL,
         CI_PROJECT_DIR,
         CI_COMMIT_BRANCH,
+        CI_COMMIT_TAG,
         CI_COMMIT_SHA,
-        CI_REPOSITORY_URL
+        CI_REPOSITORY_URL,
+        CI_JOB_URL: GITLAB_CI_JOB_URL
       } = env
-      return {
-        [BUILD_SOURCE_ROOT]: CI_PROJECT_DIR,
+
+      const tags = {
         [CI_PIPELINE_ID]: GITLAB_PIPELINE_ID,
         [CI_PIPELINE_NAME]: CI_PROJECT_PATH,
         [CI_PIPELINE_NUMBER]: CI_PIPELINE_IID,
-        [CI_PIPELINE_URL]: `${(CI_PIPELINE_URL || '').replace('/-/pipelines/', '/pipelines/')}`,
         [CI_PROVIDER_NAME]: 'gitlab',
-        [CI_WORKSPACE_PATH]: CI_PROJECT_DIR,
-        [GIT_BRANCH]: CI_COMMIT_BRANCH,
         [GIT_COMMIT_SHA]: CI_COMMIT_SHA,
-        [GIT_REPOSITORY_URL]: CI_REPOSITORY_URL
+        [DEPRECATED_GIT_COMMIT_SHA]: CI_COMMIT_SHA,
+        [GIT_REPOSITORY_URL]: filterSensitiveInfoFromRepository(CI_REPOSITORY_URL),
+        [CI_JOB_URL]: GITLAB_CI_JOB_URL
+      }
+
+      if (CI_COMMIT_TAG) {
+        tags[GIT_TAG] = normalizeRef(CI_COMMIT_TAG)
+      }
+      if (CI_COMMIT_BRANCH) {
+        tags[GIT_BRANCH] = normalizeRef(CI_COMMIT_BRANCH)
       }
+      if (CI_PROJECT_DIR) {
+        tags[CI_WORKSPACE_PATH] = resolveTilde(CI_PROJECT_DIR)
+      }
+      if (GITLAB_PIPELINE_URL) {
+        tags[CI_PIPELINE_URL] = `${(GITLAB_PIPELINE_URL).replace('/-/pipelines/', '/pipelines/')}`
```

Comment:
Can this be done in the agent/backend? Why not report the raw data?

Reply:
I agree it makes sense for the backend to do it. I think the rationale behind doing it in the tracers is: we *need* to distinguish between CI providers here because there is no other way to know which env vars to look for. Since we have to make that branching logic, we might as well use it to "clean up" a bit the data. 

This is part of a larger discussion though.